recirq/benchmarks/rep_rate/rep_rate_calculator.py

```diff
@@ -0,0 +1,263 @@
+    (or shot) to execute this on the device.  This rate varies widely
+    based on width, depth, and amount of sweeps/batching in the
+    circuit.  This class will help test various rates on the device.
+
+    This class is used to generate statistics on the processor datasheets
+    for latency and repetition rates.
+    """
+
+    def __init__(self,
+                 project_id: str,
+                 processor_id: str,
+                 gate_set: Optional[cg.SerializableGateSet] = None,
+                 device: Optional[cirq.Device] = None,
+                 sampler: Optional[cirq.Sampler] = None):
+        """Initialize the tester object.
+
+        Args:
+            project_id: Cloud project id as a string
+            processor_id: Processor to test on
+            gate_set: gate set to use.  Defaults to square root of iswap
+            device: device object to use. defaults to device provided by engine
+            sampler: sampler object to use.  defaults to engine sampler
+        """
+        if not device or not sampler:
+            engine = cg.Engine(project_id=project_id)
+        gate_set = gate_set or cg.SQRT_ISWAP_GATESET
+        self.device = device or engine.get_processor(processor_id).get_device(
+            gate_sets=[gate_set])
+        self.sampler = sampler or engine.sampler(processor_id=processor_id,
+                                                 gate_set=gate_set)
+        self.gate = cirq.ISWAP**0.5
+        self.qubits = list(self.device.qubits)
+        if gate_set == cg.SYC_GATESET:
+            self.gate = cirq.google.SYC
+
+        # log of print statements for testing
+        self.print_log = ''
+
+    def _flush_print_log(self) -> None:
+        """Remove print log used fro debugging."""
+        self.print_log = ''
+
+    def _print(self, s: str) -> None:
+        """Includes printing to testing log and flushing."""
+        self.print_log += s
+        self.print_log += '\n'
+        print(s, flush=True)
+
+    def _latency_circuit(self) -> cirq.Circuit:
+        """Circuit for measuring round-trip latency.
+
+        Use the most simple circuit possible, a measurement of a single
+        random qubit.
+        """
+        return cirq.Circuit(cirq.measure(random.choice(self.qubits)))
+
+    def _entangling_layer(self, qubits: List[cirq.Qid]) -> cirq.Moment:
+        """Creates a layer of random two-qubit gates."""
+        m = cirq.Moment()
+        for q in qubits:
+            if q in m.qubits:
+                continue
+            pairings = [
+                q + adj
+                for adj in [(0, 1), (1, 0), (0, -1), (-1, 0)]
+                if q + adj in qubits and q + adj not in m.qubits
+            ]
+            if not pairings:
+                continue
+            q2 = random.choice(pairings)
+            m = m.with_operation(self.gate(q, q2))
+        return m
+
+    def _sq_layer(self, qubits: List[cirq.Qid], parameterized: bool,
```

Comment:
this can be a free function or should be annotated with `@staticmethod`

Reply:
Done